http/parse.go

```diff
@@ -61,22 +61,28 @@ func parseRequest(r *http.Request, root *cmds.Command) (*cmds.Request, error) {
 	}
 
 	opts, stringArgs2 := parseOptions(r)
+	iopts := make(map[string]interface{}, len(opts))
 	optDefs, err := root.GetOptions(pth)
 	if err != nil {
 		return nil, err
 	}
 	for k, v := range opts {
+		iopts[k] = v
 		if optDef, ok := optDefs[k]; ok {
 			name := optDef.Names()[0]
 			if k != name {
-				opts[name] = v
-				delete(opts, k)
+				iopts[name] = v
+				delete(iopts, k)
+			}
+
+			if optDef.Type() != cmds.Strings && len(v) > 0 {
+				iopts[name] = v[0]
```

Comment:
Should we set `iopts[name]` to empty if `len(v) == 0`?

Reply:
We should probably fail if we get something we don't expect. That includes an empty value (should be impossible) and multiple values where exactly one is expected.

Reply:
done. although it turns out `len(v)==0` isn't a thing as `URL.Query()` just returns `[]string{""}` if there are no values.